workloads/validation.rn

```diff
@@ -0,0 +1,153 @@
+
+use latte::*;
+
+const ROW_COUNT = latte::param!("row_count", 1000000);
+const OFFSET = latte::param!("offset", 0);
+const REPLICATION_FACTOR = latte::param!("replication_factor", 3);
+const ROWS_PER_PARTITION = latte::param!("rows_per_partition", 1);
+// NOTE: 'partition_sizes' defines set of 'percent:multiplier' pairs to create multi-row partitions
+// of different sizes. Example: '95:1,4:2,1:4'
+const PARTITION_SIZES = latte::param!("partition_sizes", "100:1");
+
+const USE_PREPARED_STATEMENTS = latte::param!("use_prepared_statements", true);
+
+const KEYSPACE = "latte";
+const TABLE = "validation";
+
+const P_STMT = #{
+    "INSERT": #{
+        "NAME": "p_stmt_validation__insert",
+        "CQL": `INSERT INTO ${KEYSPACE}.${TABLE}(pk, ck) VALUES (:pk, :ck)`,
+    },
+    "GET": #{
+        "NAME": "p_stmt_validation__get",
+        "CQL": `SELECT pk, ck FROM ${KEYSPACE}.${TABLE} WHERE pk = :pk LIMIT 1`,
+    },
+    "GET_MANY": #{
+        "NAME": "p_stmt_validation__get_many",
+        "CQL": `SELECT pk, ck FROM ${KEYSPACE}.${TABLE} WHERE pk = :pk LIMIT :max_limit`,
+    },
+    "COUNT": #{
+        "NAME": "p_stmt_validation__count",
+        "CQL": `SELECT COUNT(*) FROM ${KEYSPACE}.${TABLE} WHERE pk = :pk`,
+    },
+};
+
+pub async fn schema(db) {
+    db.execute(`CREATE KEYSPACE IF NOT EXISTS ${KEYSPACE} WITH REPLICATION = {
+        'class': 'NetworkTopologyStrategy', 'replication_factor': ${REPLICATION_FACTOR} }`).await?;
+
+    db.execute(`CREATE TABLE IF NOT EXISTS ${KEYSPACE}.${TABLE}(
+        pk bigint,
+        ck bigint,
+        PRIMARY KEY (pk, ck)
+    ) WITH CLUSTERING ORDER BY (ck ASC)`).await?;
+}
+
+pub async fn erase(db) {
+    db.execute(`TRUNCATE TABLE ${KEYSPACE}.${TABLE}`).await?
+}
+
+pub async fn prepare(db) {
+    db.init_partition_row_distribution_preset(
+        "main", ROW_COUNT, ROWS_PER_PARTITION, PARTITION_SIZES,
+    ).await?;
+    db.prepare(P_STMT.INSERT.NAME, P_STMT.INSERT.CQL).await?;
+    db.prepare(P_STMT.GET.NAME, P_STMT.GET.CQL).await?;
+    db.prepare(P_STMT.GET_MANY.NAME, P_STMT.GET_MANY.CQL).await?;
+    db.prepare(P_STMT.COUNT.NAME, P_STMT.COUNT.CQL).await?;
+}
+
+// User functions
+
+pub async fn insert(db, i) { // validation is not applicable
+    let idx = i % ROW_COUNT + OFFSET;
+    let partition = db.get_partition_info("main", idx).await;
+    partition.idx += OFFSET;
+    let pk = hash(partition.idx);
+    let ck = hash(idx);
+    if USE_PREPARED_STATEMENTS {
+        db.execute_prepared(P_STMT.INSERT.NAME, [pk, ck]).await?
+    } else {
+        db.execute(P_STMT.INSERT.CQL.replace(":pk", `${pk}`).replace(":ck", `${ck}`)).await?
+    }
+}
+
+pub async fn get(db, i) { // make sure that we have only 1 row no matter how big partitions
+    let idx = i % ROW_COUNT + OFFSET;
+    let partition = db.get_partition_info("main", idx).await;
+    partition.idx += OFFSET;
+    let pk = hash(partition.idx);
+    let custom_err = "custom very useful err msg for the 'get' function"; // optional
```

Comment:
let's try putting something meaningful in this example ? i.e.

"expected to read only one row"

also would be nice to put in a comment here or on the readme, how would it gonna be presented when it would be failing

Reply:
Ok, will do in a separate PR.

Reply:
@fruch 
- https://github.com/scylladb/latte/pull/64/files